### PR TITLE
[RBFDD-451] Document secure Flink Kafka connector

### PR DESCRIPTION
**Pull request by stefanobaghino**
- [RBFDD-449] Document Flink SBT
- [RBFDD-451] Document secure Flink Kafka connector

### CODE REVIEW

File: flink/flink.rst [hunk 1]
```diff
@@ -6,12 +6,10 @@ Flink
 
 The following documentation will target RBP-specific features: where not else stated, please refer to the `official documentation <https://ci.apache.org/projects/flink/flink-docs-release-1.0/>`_.
```
Comment (giampaolotrapasso) [inline flink/flink.rst:7]: RBP should be RBFDD

File: flink/sbt/sbt.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
+===============================
+Using SBT to create a Flink job
+===============================
+
+Motivation
+----------
+
+Maven is a widely accepted build tool for projects that target the Java platform. The Radicalbit Fast Data Distribution provides an archetype to kickstart your development projects using Maven, but Scala developers may find more compelling to use the Maven alternative of choice for the Scala ecosystem, SBT (Simple Build Tool).
```
Comment (giampaolotrapasso) [inline flink/sbt/sbt.rst]: Maven/SBT links would improve doc. Spitting into two sentences may improve readability.

File: flink/sbt/sbt.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+Basic 'archetype'
+-----------------
+
+Currently, SBT does not provide built-in capabilities to scaffold a project like Maven archetypes do.
```
Comment (giampaolotrapasso) [inline flink/sbt/sbt.rst]: With possibility of creating more examples in Scala, maybe we want to evaluate [this plugin](https://github.com/softprops/np#for-sbt-013) that allows with a single line in sbt configuration, to create an empty Scala project.

File: flink/sbt/sbt.rst [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    mkdir my-flink-project
+
+Then, move into the directory and create a project directory::
+
+    cd my-flink-project
+    mkdir project
+
+To package our application, we'll use the ``sbt-assembly`` plugin. To add it to our build, create and open in your favorite text editor the ``project/assembly.sbt`` file and fill it in with this single line:
+
+.. literalinclude:: /rbp-docs-code/sbt/project/assembly.sbt
```
Comment (giampaolotrapasso) [inline flink/sbt/sbt.rst]: Something is not working with modules. Checked out branch, rebased and build doc. However it does not load referenced code. Could you verify?

File: flink/sbt/sbt.rst [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+===============================
+
+Motivation
+----------
+
+Maven is a widely accepted build tool for projects that target the Java platform. The Radicalbit Fast Data Distribution provides an archetype to kickstart your development projects using Maven, but Scala developers may find more compelling to use the Maven alternative of choice for the Scala ecosystem, SBT (Simple Build Tool).
+
+Basic 'archetype'
+-----------------
+
```
Comment (giampaolotrapasso) [inline flink/sbt/sbt.rst]: Why not giving user the url of the code? Probably is more simple to checkout that repo and run the provided example. Explanations of steps as below can follow.

**Comment by stefanobaghino** [comment]
@giampaolotrapasso Open for review, it now only refers to the Flink Kafka secure connector.

File: flink/connectors/kafka.rst [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+Kafka Connector
+===============
+
+Due to its nature, the distributed publish/subscribe Apache Kafka is one of the best matches to read data from and write data to for Flink.
+
+The `official documentation <https://ci.apache.org/projects/flink/flink-docs-release-1.0/apis/streaming/connectors/kafka.html>`_ contains in-depth details on the high level of interoperability between Kafka and Flink.
+
+To use the connector, you should first of all include the following dependency in your project's POM:
+
+.. literalinclude:: /rbp-docs-code/mvn/pom.xml
```
Comment (giampaolotrapasso) [inline flink/connectors/kafka.rst]: This is how the pox gets rendered on the branch.
schermata 2016-05-17 alle 10 15 21

File: flink/connectors/kafka.rst [hunk 7]
```diff
@@ -0,0 +1,28 @@
+===============
+Kafka Connector
```
Comment (giampaolotrapasso) [inline flink/connectors/kafka.rst:2]: Please as Akka/Cassandra section, introduce more structure into the content

File: flink/connectors/kafka.rst [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+The `official documentation <https://ci.apache.org/projects/flink/flink-docs-release-1.0/apis/streaming/connectors/kafka.html>`_ contains in-depth details on the high level of interoperability between Kafka and Flink.
+
+To use the connector, you should first of all include the following dependency in your project's POM:
+
+.. literalinclude:: /rbp-docs-code/mvn/pom.xml
+    :caption: pom.xml
+    :language: xml
+    :lines: 124-128
+
+RBFDD includes a custom Flink version that allows the user to make the two systems interoperate even on *secure* clusters.
```
Comment (giampaolotrapasso) [inline flink/connectors/kafka.rst:22]: I would introduce a _Security_ section here, to state better that connector can be used with and _without_ Kerberos security if I understood right that Kerberos security is not mandatory.

File: flink/connectors/kafka.rst [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+The feature is mostly transparent to the client code, the only required step is to add a property to the ``FlinkKafkaConsumer09`` and/or the ``FlinkKafkaProducer09``, this property being ``security.protocol``, whose value should be set to ``SASL_PLAINTEXT``.
+
+.. Note::
+
+    RBFDD ships with Kafka |kafka-version|, which supports Kerberos security. Please be advised that producers and consumers for previous versions of Kafka, despite being shipped along with Flink, *won't work* due to wire-protocol incompatibility. You should always use the aforementioned ``FlinkKafkaConsumer09`` and ``FlinkKafkaProducer09``.
+
+The following program serves as an example of a trivial job using Kafka as both source and sink that can run on a secure cluster as well.
+
+.. literalinclude:: /rbp-docs-code/sbt/src/main/scala/io/radicalbit/flink/kafka/KafkaExample.scala
```
Comment (giampaolotrapasso) [inline flink/connectors/kafka.rst]: Something is wrong with modules, it does not link the code

**Comment by giampaolotrapasso** [comment]
Thanks

**Pull request merged**